moveit_demo_nodes/run_move_group/launch/run_move_group.launch.py

```diff
@@ -1,115 +1,37 @@
-    )
-    robot_description_kinematics = {"robot_description_kinematics": kinematics_yaml}
-
-    # Planning Functionality
-    ompl_planning_pipeline_config = {
-        "planning_pipelines": ["ompl"],
-        "ompl": {
-            "planning_plugin": "ompl_interface/OMPLPlanner",
-            "request_adapters": """default_planner_request_adapters/AddTimeOptimalParameterization default_planner_request_adapters/FixWorkspaceBounds default_planner_request_adapters/FixStartStateBounds default_planner_request_adapters/FixStartStateCollision default_planner_request_adapters/FixStartStatePathConstraints""",
-            "start_state_max_bounds_error": 0.1,
-        },
-    }
-    ompl_planning_yaml = load_yaml(
-        "moveit_resources_panda_moveit_config", "config/ompl_planning.yaml"
-    )
-    ompl_planning_pipeline_config["ompl"].update(ompl_planning_yaml)
-
-    # Trajectory Execution Functionality
-    moveit_simple_controllers_yaml = load_yaml(
-        "moveit_resources_panda_moveit_config", "config/panda_controllers.yaml"
+    moveit_config = (
+        MoveItConfigsBuilder("moveit_resources_panda")
+        .robot_description(file_path="config/panda.urdf.xacro")
+        .robot_description_semantic()
+        .robot_description_kinematics()
+        .joint_limits()
+        .trajectory_execution(file_path="config/panda_gripper_controllers.yaml")
+        .planning_scene_monitor()
+        .planning_pipelines()
+        .moveit_configs()
     )
-    moveit_controllers = {
-        "moveit_simple_controller_manager": moveit_simple_controllers_yaml,
-        "moveit_controller_manager": "moveit_simple_controller_manager/MoveItSimpleControllerManager",
-    }
-
-    trajectory_execution = {
-        "moveit_manage_controllers": True,
-        "trajectory_execution.allowed_execution_duration_scaling": 1.2,
-        "trajectory_execution.allowed_goal_duration_margin": 0.5,
-        "trajectory_execution.allowed_start_tolerance": 0.01,
-    }
-
-    planning_scene_monitor_parameters = {
-        "publish_planning_scene": True,
-        "publish_geometry_updates": True,
-        "publish_state_updates": True,
-        "publish_transforms_updates": True,
-    }
-
-    joint_limits_yaml = {
-        "robot_description_planning": load_yaml(
-            "moveit_resources_panda_moveit_config", "config/joint_limits.yaml"
-        )
-    }
 
     # Start the actual move_group node/action server
     run_move_group_node = Node(
         package="moveit_ros_move_group",
         executable="move_group",
         output="screen",
         parameters=[
-            robot_description,
-            robot_description_semantic,
-            kinematics_yaml,
-            ompl_planning_pipeline_config,
-            trajectory_execution,
-            moveit_controllers,
-            planning_scene_monitor_parameters,
-            joint_limits_yaml,
+            moveit_config.robot_description,
```

Comment:
It would be nice if you could just write:

```python
parameters = [moveit_config.to_dict()]
```
or something like that?

Reply:
I'll update it now